troopers.sh

```diff
@@ -1,76 +1,89 @@
 #!/bin/bash
 
-prefix="`dirname $0`/"          # prefix should be with "/" in the end
+trap "cleanup" EXIT
+
 login=$1                        # 1st argument of cli
 password=$2                     # 2nd argument of cli
-curl_opt="-s -b ${prefix}cookie.$login -c ${prefix}cookie.$login"
-exit_cycle=0
-selected_enemy=1                # if you want fight with specific avatar
 friend=$3
-if [ -z "$friend" ]
-then
-  friend="roushet"             # avatar's name
-fi
+site="http://$login.minitroopers.com"
+cookie_file="$(mktemp -t "$login.XXXXXX" --suffix='.cookie')"
+curl="curl --silent --cookie "$cookie_file""
+egrep="grep --extended-regexp --only-matching"
 
 # Check for "raids"
-function check {
-    message=`egrep "(Another|Shortage)" ${prefix}index`
-    if [ "$message" == "Shortage" ] || [ -z "$message" ]
-    then
-        exit_cycle=1
-        mission
-    fi
-}
-
-# Get money
-function getmoney {
-    money=`grep money ${prefix}index -A1|tail -n1`
-    echo "$login has earned $money coins"
+function hasRecruits {
+    $curl "$site/hq" | grep "Another" > /dev/null
+    return $?
```

Comment:
Getting rid of all intermediate files and piping `curl`s directly into `grep`s was a necessary step to obtain coherent and predictable behaviour with parallel tasks.